api/logical.go

```diff
@@ -65,6 +65,16 @@ func (c *Logical) ReadWithDataWithContext(ctx context.Context, path string, data
 	ctx, cancelFunc := c.c.withConfiguredTimeout(ctx)
 	defer cancelFunc()
 
+	// In order to support responses with both top-level data
+	// and no top-level data objects, get the raw response
+	// If no top-level data object from api response, we add
+	// any existing raw data from api response in data part of
+	// the secret to be returned
+	// This would return responses for vault read with or without
+	// -format=raw
+	data = make(map[string][]string)
+	data["-format"] = []string{"raw"}
+
```

Comment:
This seems like it mimics the `-format` CLI flag but I don't think the API makes use of that in anyway. I tried `vault read sys/health` with these lines commented out and it seemed to work as expected per the PR description. With that said, I think we should probably remove this.

Reply:
I initially added this and the raw response somehow worked so I thought this was needed 😄 . It works without this and is not needed. I will remove this.